src/saturn_engine/worker/executors/arq/executor.py

```diff
@@ -48,13 +59,49 @@ async def process_message(self, message: ExecutableMessage) -> PipelineResults:
         config = self.config.load_object(message.config)
         options = config.cast_namespace(ARQ_EXECUTOR_NAMESPACE, ARQExecutor.Options)
 
-        job = await (await self.redis_queue).enqueue_job(
-            EXECUTE_FUNC_NAME,
-            message.message,
-            _expires=options.queue_timeout,
-            _queue_name=options.queue_name,
+        try:
+            job = await (await self.redis_queue).enqueue_job(
+                EXECUTE_FUNC_NAME,
+                message.message,
+                _expires=options.queue_timeout,
+                _queue_name=options.queue_name,
+            )
+        except (OSError, aioredis.errors.ConnectionClosedError):
```

Comment:
Also fix a bug here where a redis disconnection would end up in an infinite failure loop..